python/revng/daemon/__init__.py

```diff
@@ -34,7 +36,23 @@
 DEBUG = config("STARLETTE_DEBUG", cast=bool, default=False)
 
 
-def get_middlewares() -> List[Middleware]:
+class ManagerCredentialsMiddleware:
+    def __init__(self, app: ASGIApp, event_manager: EventManager):
+        self.app = app
+        self.event_manager = event_manager
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if scope["type"] != "http":
+            return await self.app(scope, receive, send)
+
+        credentials = Headers(scope=scope).get("x-revng-set-credentials")
```

Comment:
What's the convention about custom headers naming?
I'd expect `X-revng-Set-Credentials` or something like that. What's the best practice here?

Reply:
Headers are case-insensitive ([Reference](https://datatracker.ietf.org/doc/html/rfc9112#name-field-syntax))

Most implementations that I know of use `.lower` when parsing headers

Reply:
Yeah, it was more for aesthetic purposes, no biggie.